src/machine/machinectl.c

```diff
@@ -2158,8 +2159,10 @@ static int export_tar(int argc, char *argv[], void *userdata) {
         path = empty_or_dash_to_null(path);
 
         if (path) {
-                if (arg_format == IMPORT_COMPRESS_UNKNOWN)
-                        arg_format = tar_filename_to_compression(path);
+                if (strempty(arg_format)) {
+                        free(arg_format);
+                        arg_format = strdup(import_compress_type_to_string(tar_filename_to_compression(path)));
+                }
```

Comment:
why this patch? we typically want that stuff if parsed early and passed around in parsed form internally, and only converted to non-parsed versions when we pass it on externally.

hence, maybe define a helper somewhere that parses the compression format and the compression level from a single string, and returns both, plus another helper that formats it back into a string. Then use that here.

(also this lacks oom check)